fedbiomed/researcher/secagg/_secagg_context.py

```diff
@@ -158,11 +161,10 @@ def set_experiment_id(self, experiment_id: Union[str, None]) -> None:
             FedbiomedSecaggError: bad argument type or value
         """
 
-        if not isinstance(experiment_id, (str, type(None))):
-            errmess = f'{ErrorNumbers.FB415.value}: bad parameter `experiment_id` must be a str or None if the ' \
-                      f'context is set for biprime.'
-            logger.error(errmess)
-            raise FedbiomedSecaggError(errmess)
+        if not isinstance(experiment_id, str):
```

Comment:
Misc: please update accordingly the prototype and docstrings including for:
- SecaggContext.__init__()
- SecaggContext.experiment_id()
- SecaggContext.set_experiment_id()
- SecaggMpspdzContext.__init__()

Reply:
`experiment_id` type `Union[str, None]` was existing specifically for the case of biprime. For server key and DH, they are always associated to one experiment thus `experiment_id` type should now always be `str`

We should also propagate that to:
- `fedbiomed.researcher.secure_aggregation` (this is only argument type/docstrings/class attribute type updates)